lib/prometheus_exporter/middleware.rb

```diff
@@ -6,12 +6,38 @@
 class PrometheusExporter::Middleware
   MethodProfiler = PrometheusExporter::Instrumentation::MethodProfiler
 
+  module RedisInstrumenter
+    def call(_command, _config)
+      unless prof = Thread.current[:_method_profiler]
+        super
+      end
+
+      start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+      super
+    ensure
+      data = (prof[:redis] ||= { duration: 0.0, calls: 0 })
+      data[:duration] += Process.clock_gettime(Process::CLOCK_MONOTONIC) - start
+      data[:calls] += 1
+    end
+
+    alias_method :call_pipelined, :call
```

Comment:
This will cause issues. If you have other middleware registered, you'll cause `call_pipelined` events to become `call` events.

I know it's a bit unfortunate, I only realized this after the API was published.

Reply:
oh yikes... will see if I can add a test for this, thanks for catching it